### ISSUE TITLE
Biome false positive on a tag used for navigation.

### ISSUE THREAD

**Issue by NyanHelsing** [biomejs/biome#1873]
### Environment information

```bash
I have something like:



  ✖ Use a button element instead of an a element.

     6 │     return (
     7 │         <a
   > 8 │             onClick={(event) => {
       │             ^^^^^^^^^^^^^^^^^^^^^
   > 9 │                 event.preventDefault();
  > 10 │                 window.history.pushState({}, '', to);
  > 11 │             }}
       │             ^^
    12 │             {...props}
    13 │         >

  ℹ Anchor elements should only be used for default sections or page navigation

  ℹ Check this thorough explanation to better understand the context.

```

which is a false positive because this *is* being used for page navigation.
```


### Rule name

lint/a11y/useValidAnchor

### Playground link

https://biomejs.dev/playground/?files.main.tsx=aQBtAHAAbwByAHQAIABSAGUAYQBjAHQAIABmAHIAbwBtACAAIgByAGUAYQBjAHQAIgA7AAoAaQBtAHAAbwByAHQAIAB7ACAAdQBzAGUATABvAGMAYQB0AG8AcgAgAH0AIABmAHIAbwBtACAAIgAuAC8AbABvAGMAYQB0AG8AcgAuAGoAcwB4ACIAOwAKAAoACgBlAHgAcABvAHIAdAAgAGMAbwBuAHMAdAAgAEwAaQBuAGsAIAA9ACAAKAB7ACAAdABvACwAIABjAGgAaQBsAGQAcgBlAG4ALAAgAC4ALgAuAHAAcgBvAHAAcwAgAH0AKQAgAD0APgAgAHsACgAgACAAIAAgAGMAbwBuAHMAdAAgAGwAbwBjAGEAdABvAHIAIAA9ACAAdQBzAGUATABvAGMAYQB0AG8AcgAoACkAOwAKACAAIAAgACAAcgBlAHQAdQByAG4AIAAoAAoAIAAgACAAIAAgACAAIAAgADwAYQAKACAAIAAgACAAIAAgACAAIAAgACAAIAAgAG8AbgBDAGwAaQBjAGsAPQB7ACgAZQB2AGUAbgB0ACkAIAA9AD4AIAB7AAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIABlAHYAZQBuAHQALgBwAHIAZQB2AGUAbgB0AEQAZQBmAGEAdQBsAHQAKAApADsACgAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgAHcAaQBuAGQAbwB3AC4AaABpAHMAdABvAHIAeQAuAHAAdQBzAGgAUwB0AGEAdABlACgAewB9ACwAIAAiACIALAAgAHQAbwApADsACgAgACAAIAAgACAAIAAgACAAIAAgACAAIAB9AH0ACgAgACAAIAAgACAAIAAgACAAIAAgACAAIAB7AC4ALgAuAHAAcgBvAHAAcwB9AAoAIAAgACAAIAAgACAAIAAgAD4ACgAgACAAIAAgACAAIAAgACAAIAAgACAAIAB7AGMAaABpAGwAZAByAGUAbgB9AAoAIAAgACAAIAAgACAAIAAgADwALwBhAD4ACgAgACAAIAAgACkAOwAKAH0AOwAKAA%3D%3D

### Expected result

This should be detected as a use of the anchor tag for page navigation.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by togami2864** [comment]
I don't think we should treat that code as a navigation. While it's true that the click event directs to a navigation URL, assistive technologies such as screen readers do not recognize a tag without 'href' attribute as a navigational link
